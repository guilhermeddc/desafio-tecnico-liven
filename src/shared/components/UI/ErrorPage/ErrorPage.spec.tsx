import {QueryClient, QueryClientProvider, QueryObserver} from 'react-query';

import {fireEvent, render, screen} from '@testing-library/react';

import {ErrorPage} from '.';

describe('ErrorPage', () => {
  it('renders correctly', () => {
    const myQueryFn = jest.fn();

    const queryClient = new QueryClient();

    new QueryObserver(queryClient, {
      queryKey: 'products',
    });
    const {container} = render(
      <QueryClientProvider client={queryClient}>
        <ErrorPage queryKey="" />
      </QueryClientProvider>,
    );

    const unsubscribeQuery = queryClient.getQueryCache().subscribe((query) => {
      if (String(query?.query) === 'products') {
        myQueryFn();
      }
    });

    expect(myQueryFn).toHaveBeenCalledTimes(0);
    unsubscribeQuery();

    fireEvent.click(screen.getByRole('button'));
    unsubscribeQuery();
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    const queryClient = new QueryClient();

    render(
      <QueryClientProvider client={queryClient}>
        <ErrorPage queryKey="" />
      </QueryClientProvider>,
    );

    expect(screen.getByText('Error loading data!')).toBeInTheDocument();
    expect(screen.getByText('Reset Page')).toBeInTheDocument();
  });
});
