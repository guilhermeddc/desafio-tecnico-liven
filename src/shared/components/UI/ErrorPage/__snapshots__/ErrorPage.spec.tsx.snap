// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders correctly 1`] = `
<div>
  <div
    class="css-184kej8-MuiStack-root"
  >
    <div
      class="css-1k6lqaj-MuiStack-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-16tkpzi-MuiSvgIcon-root"
        data-testid="ErrorRoundedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 11c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
        />
      </svg>
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-149xwtl-MuiTypography-root"
      >
        Error loading data!
      </h4>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Reset Page
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
